File: QHSEAnalytics/src/main/java/com/QHSEAnalytics/dto/response/KpiResponse.java
package com.QHSEAnalytics.dto.response;

import com.QHSEAnalytics.entity.UniteKpi;
import com.QHSEAnalytics.enums.Direction;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class KpiResponse {
    private Long id;
    private String nom;
    private String definition;
    private UniteKpi unite;
    private String categorieCode;
    private String categorieLibelle;
    private Double seuilFaible;
    private Double seuilModere;
    private Double seuilCritique;
    private Integer ordre;
    private boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Direction direction;
    private Double targetValue;
}

File: qhse-frontend/src/app/shared/models/kpi-response.ts
import { UniteKpi } from '../enums/unite-kpi.enum';

export interface KpiResponse {
  id: number;
  nom: string;
  definition: string;
  unite: UniteKpi;
  categorieCode: string;
  categorieLibelle: string;
  seuilFaible: number;
  seuilModere: number;
  seuilCritique: number;
  ordre: number;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}
